atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/KvsSweepProgressQueueTest.java

```diff
@@ -37,86 +37,73 @@ public void setup() {
-        progress.numberOfShards();
         progress.updateNumberOfShards(128);
-        assertThat(progress.numberOfShards()).isEqualTo(128);
+        assertThat(progress.getNumberOfShards()).isEqualTo(128);
     }
 
     @Test
     public void attemptingToDecreaseNumberOfShardsIsNoop() {
-        progress.numberOfShards();
         progress.updateNumberOfShards(64);
         progress.updateNumberOfShards(32);
-        assertThat(progress.numberOfShards()).isEqualTo(64);
+        assertThat(progress.getNumberOfShards()).isEqualTo(64);
     }
 
     @Test
     public void increasingNumberOfShardsAbove128Throws() {
-        progress.numberOfShards();
         assertThatThrownBy(() -> progress.updateNumberOfShards(256)).isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
     public void canReadInitialSweptTimestamp() {
-        assertThat(progress.lastSweptTimestamp(10, true)).isEqualTo(0L);
+        assertThat(progress.getLastSweptTimestampPartition(10, true)).isEqualTo(-1L);
     }
 
     @Test
     public void canUpdateSweptTimestamp() {
-        progress.lastSweptTimestamp(10, true);
-        progress.updateLastSweptTimestamp(10, true, 1024L);
-        assertThat(progress.lastSweptTimestamp(10, true)).isEqualTo(1024L);
+        progress.updateLastSweptTimestampPartition(10, true, 1024L);
+        assertThat(progress.getLastSweptTimestampPartition(10, true)).isEqualTo(1024L);
     }
 
     @Test
     public void attemptingToDecreaseSweptTimestampIsNoop() {
-        progress.lastSweptTimestamp(10, true);
-        progress.updateLastSweptTimestamp(10, true, 1024L);
-        progress.updateLastSweptTimestamp(10, true, 512L);
-        assertThat(progress.lastSweptTimestamp(10, true)).isEqualTo(1024L);
+        progress.updateLastSweptTimestampPartition(10, true, 1024L);
+        progress.updateLastSweptTimestampPartition(10, true, 512L);
+        assertThat(progress.getLastSweptTimestampPartition(10, true)).isEqualTo(1024L);
     }
 
     @Test
     public void updatingTimestampForOneShardDoesNotAffectOthers() {
-        assertThat(progress.lastSweptTimestamp(10, true)).isEqualTo(0L);
-        assertThat(progress.lastSweptTimestamp(20, true)).isEqualTo(0L);
-        progress.updateLastSweptTimestamp(10, true, 1024L);
-        assertThat(progress.lastSweptTimestamp(20, true)).isEqualTo(0L);
-
-        progress.updateLastSweptTimestamp(20, true, 512L);
-        assertThat(progress.lastSweptTimestamp(20, true)).isEqualTo(512L);
-        assertThat(progress.lastSweptTimestamp(10, true)).isEqualTo(1024L);
+        progress.updateLastSweptTimestampPartition(10, true, 1024L);
+        assertThat(progress.getLastSweptTimestampPartition(20, true)).isEqualTo(-1L);
+
+        progress.updateLastSweptTimestampPartition(20, true, 512L);
+        assertThat(progress.getLastSweptTimestampPartition(20, true)).isEqualTo(512L);
+        assertThat(progress.getLastSweptTimestampPartition(10, true)).isEqualTo(1024L);
```

Comment:
nit: maybe make this more than `1024L` (since the second update could have hit (10, true) and it might have been a noop)